list.rb

```diff
@@ -10,4 +10,4 @@ def initialize(*elements)
   def each(&block)
     @list.each(&block)
   end
-end
```

Comment:
- [ ] [OPTIONAL] For your constructor, I see that you implemented like below 👇🏾 

### Your approach
```ruby
def initialize(*elements)
    @list = elements
  end
```

But I would suggest that you used this approach. 👇🏾 

### My suggestion
```ruby
def initialize(*args)
        @list = []
        args.each { |arg| @list << arg }
    end
```

I notice a few issues with **your approach**. 

- The above constructor method directly assigns the `*elements` array to the `@list` instance variable. While this can be more efficient as it avoids the extra iteration, it also means that any subsequent modifications to the `@list` array within the class will affect the original array passed as an argument. This can lead to unexpected behavior if the original array is modified outside the class. This may not be much of an issue now but it can cause unexpected results.

- Limitation to Arrays: Using the splat operator with a parameter name like `*elements` implies that the method expects an array or multiple arguments. **_If you try to pass a single argument that is not an array, it will result in a syntax error. This limitation may reduce flexibility in some cases where you might want to pass a single value as an argument._**. This is the main issue. 

Fixing this is **optional** and will not affect the approval of this project.